internal/mux/mux.go

```diff
@@ -239,41 +238,31 @@ func (p *playerImpl) setBufferSize(bufferSize int) {
 	p.m.Lock()
 	defer p.m.Unlock()
 
-	orig := p.bufferSize
 	p.bufferSize = bufferSize
 	if bufferSize == 0 {
 		p.bufferSize = p.mux.defaultBufferSize()
 	}
-	if orig != p.bufferSize {
-		p.bufPool = nil
-	}
 }
 
-func (p *playerImpl) getTmpBuf() ([]byte, func()) {
-	// The returned buffer could be accessed regardless of the mutex m (#254).
-	// In order to avoid races, use a sync.Pool.
-	// On the other hand, the calls of getTmpBuf itself should be protected by the mutex m,
-	// then accessing p.bufPool doesn't cause races.
-	if p.bufPool == nil {
-		p.bufPool = &sync.Pool{
-			New: func() interface{} {
-				buf := make([]byte, p.bufferSize)
-				return &buf
-			},
-		}
-	}
-	buf := p.bufPool.Get().(*[]byte)
-	return *buf, func() {
-		// p.bufPool could be nil when setBufferSize is called (#258).
-		// buf doesn't have to (or cannot) be put back to the pool, as the size of the buffer could be changed.
-		if p.bufPool == nil {
-			return
-		}
-		if len(*buf) != p.bufferSize {
-			return
-		}
-		p.bufPool.Put(buf)
+var bufPool = sync.Pool{
+	New: func() interface{} {
+		// we do not know the intended size here, but creating a new slice should
+		// be the cold path anyways
+		buf := make([]byte, 0)
+		return &buf
+	},
+}
+
+func (p *playerImpl) pooledBuf() *[]byte {
```

Comment:
This doesn't have to be a `playerImpl`'s method. What about `func pooledBuf(size) *[]byte`?